applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/adjoint_structural_response_function.cpp

```diff
@@ -0,0 +1,942 @@
+            const std::string output_label = label + "_SENSITIVITY";
+
+            if (KratosComponents<Variable<double>>::Has(label) == true)
+            {
+                const Variable<double>& r_variable =
+                    KratosComponents<Variable<double>>::Get(label);
+
+                //check for output variable to save later the computed sensitivity
+                if( KratosComponents<Variable<double>>::Has(output_label) == true )
+                {
+                    const Variable<double>& r_output_variable =
+                        KratosComponents<Variable<double>>::Get(output_label);
+                    this->UpdateElementSensitivities(r_variable, r_output_variable);
+                }
+                else
+                    KRATOS_ERROR << "Unsupported element variable for output: " << output_label << "." << std::endl;
+
+
+            }
+            else if (KratosComponents<Variable<array_1d<double,3>>>::Has(label) == true)
+            {
+                const Variable<array_1d<double,3>>& r_variable =
+                    KratosComponents<Variable<array_1d<double,3>>>::Get(label);
+
+                //check for output variable to save later the computed sensitivity
+                if (KratosComponents<Variable<array_1d<double,3>>>::Has(output_label) == true)
+                {
+                    const Variable<array_1d<double,3>>& r_output_variable =
+                        KratosComponents<Variable<array_1d<double,3>>>::Get(output_label);
+                    this->UpdateElementSensitivities(r_variable, r_output_variable);
+                }
+                else
+                    KRATOS_ERROR << "Unsupported element variable for output: " << output_label << "." << std::endl;
+
+
+            }
+            else
+                KRATOS_ERROR << "Unsupported element variable: " << label << "." << std::endl;
+        }
+
+        for (const auto& label : mConditionSensitivityVariables)
+        {
+            //create label for output variable
+            const std::string output_label = label + "_SENSITIVITY";
+
+            if (KratosComponents<Variable<double>>::Has(label) == true)
+            {
+                const Variable<double>& r_variable =
+                    KratosComponents<Variable<double>>::Get(label);
+
+                //check for output variable to save later the computed sensitivity
+                if( KratosComponents<Variable<double>>::Has(output_label) == true )
+                {
+                    const Variable<double>& r_output_variable =
+                        KratosComponents<Variable<double>>::Get(output_label);
+                    this->UpdateConditionSensitivities(r_variable, r_output_variable);
+                }
+                else
+                    KRATOS_ERROR << "Unsupported condition variable for output: " << output_label << "." << std::endl;
+            }
+            else if (KratosComponents<Variable<array_1d<double,3>>>::Has(label) == true)
```

Comment:
We might want to consider using a variadic template in a private member function to eliminate these long if-blocks of code.